src/asInput/index.jsx

```diff
@@ -49,9 +52,20 @@ const asInput = (WrappedComponent, labelFirst = true) => {
       const desc = {};
 
       if (!this.state.isValid) {
+        const hasDangerTheme = this.props.theme.includes('danger');
```

Comment:
Since the `this.props.theme.includes('danger')` check is implemented in `getDescriptions` and `render` you could consider moving this to it's own helper method and that way if you ever have to adjust it / add a theme value, you can do it in one place instead of having to remember to do it in two places.